tests/framework/test_values.py

```diff
@@ -269,27 +269,44 @@ def bad_attribute_source(index: pd.Index[int]) -> pd.DataFrame:
         pipeline(index)
 
 
-def test_attribute_pipeline_raises_no_dataframe_returned(
-    manager: ValuesManager, mocker: MockFixture
-) -> None:
-    """Test than an error is raised when something other than a pd.DataFrame is returned."""
+def test_attribute_pipeline_return_types(manager: ValuesManager, mocker: MockFixture) -> None:
     index = pd.Index([4, 8, 15, 16, 23, 42])
 
-    def bad_attribute_source(index: pd.Index[int]) -> str:
+    def series_attribute_source(index: pd.Index[int]) -> pd.Series[float]:
+        return pd.Series([1.0] * len(index), index=index)
+
+    def dataframe_attribute_source(index: pd.Index[int]) -> pd.DataFrame:
+        return pd.DataFrame({"col1": [1.0] * len(index)}, index=index)
+
+    def str_attribute_source(index: pd.Index[int]) -> str:
         return "foo"
 
-    pipeline = manager.register_attribute_producer(
-        "test_attribute", source=bad_attribute_source, component=mocker.Mock()
+    series_pipeline = manager.register_attribute_producer(
+        "test_series_attribute", source=series_attribute_source, component=mocker.Mock()
+    )
+    dataframe_pipeline = manager.register_attribute_producer(
+        "test_dataframe_attribute", source=dataframe_attribute_source, component=mocker.Mock()
+    )
+
+    assert isinstance(series_pipeline(index), pd.Series)
+    assert series_pipeline(index).index.equals(index)
+
+    assert isinstance(dataframe_pipeline(index), pd.DataFrame)
+    assert dataframe_pipeline(index).index.equals(index)
+
+    # Register the string source w/ no post-processors, i.e. calling will return str
+    bad_pipeline = manager.register_attribute_producer(
+        "test_bad_attribute", source=str_attribute_source, component=mocker.Mock()
     )
 
     with pytest.raises(
         DynamicValueError,
         match=(
-            f"The dynamic attribute pipeline for {pipeline.name} returned a {type('foo')} "
+            f"The dynamic attribute pipeline for {bad_pipeline.name} returned a {type('foo')} "
             "but pd.Series' or pd.DataFrames are expected for attribute pipelines."
         ),
     ):
-        pipeline(index)
+        bad_pipeline(index)
```

Comment:
Maybe create a pipeline with `source=str_attribute_source` and a post-processor that turns it into a Series and show that that doesn't fail?